File: CursoUdemi/C++/InferenciadeTipos/SentenciasIf.cpp
#include <iostream>
#include <string>
/*
int main() {
	using namespace std; //con esto ya no es necesario escribir std todo el rato

	const float PI = 3.14;
	float radio;
	float largo;
	float ancho;
	char opcion;
	bool hayError;

	cout << "Elige una opcion: \n\n"
		<< "a) Perimetro de rectangulo\n"
		<< "b) Area de rectangulo\n"
		<< "c) Perimetro de circunferencia \n"
		<< "d) Area de circunferencia\n\n"
		<< "x) SALIR\n\n"
		<< ">> ";

	cin >> opcion;

	switch (opcion) {
		case 'a':
		case 'A':
			cout << "Largo= ";
			cin >> largo;

			if(largo <= 0) {
				hayError = true;
			}
			else {
				hayError = false;
			}

			if (hayError) {
				cout << "ERROR: Ingresa un valor mayor que 0.";
				break;
			}
			else {
				cout << "Ancho =";
				cin >> ancho;

				hayError = ancho <= 0;

				cout << "Perimetro= " << largo * 2 + ancho * 2;
			}

			break;
		case 'b':
		case 'B':
			cout << "Largo= ";
			cin >> largo;
			cout << "Ancho =";
			cin >> ancho;

			if (largo <= 0 || ancho <= 0) {
				hayError = true;
			}
			else {
				hayError = false;
			}

			if (hayError) {
				cout << "ERROR: Ingresa un valor mayor que 0.";
				break;
			}
			else {
				cout << "Area= " << largo * ancho;
			}

			break;
		case 'c':
		case 'C':
			cout << "Radio: ";
			cin >> radio;

			if (radio < 0) {
				cout << "ERROR: Ingresa un error menor que 0";
			}
			else {
				cout << "Perimetro: " << radio * PI;
			}
			break;
		case 'd':
		case 'D':
			cout << "Radio: ";
			cin >> radio;
			if (radio < 0) {
				cout << "ERROR: Ingresa un error menor que 0";
			}
			else {
				cout << "Area: " << radio * radio * PI;
			}
			break;
		case 'x':
		case 'X':
			return 0;
		default:
			cout << "ERROR: ingresa una opcion valida";
			break;

	}

	return 0;
}

*/

File: CursoUdemi/Java/02_01SentenciasIf/src/main/java/principal/SentenciasIf.java
package principal;

import java.util.Scanner;


public class SentenciasIf {
    
    public static void main (String args[]){
        
        
    Scanner keyboard = new Scanner (System.in);
    final double PI = 3.14;
    double radio;
    double largo;
    double ancho;
    double base;
    double lado;
    double altura;
    String opcion;
    final String OPCION_1= "1";
    final String OPCION_2= "2";
    final String OPCION_3= "3";
    final String OPCION_SALIR= "0";

    System.out.println("Elige una opcion:\n\n1) o a)Rectangulo\n2) o b)Circunferencia\n"
                         + "3) o c)Triangulo\n\n 0) Salir>> ");

    opcion= keyboard.nextLine();
    
    if(!opcion.equals("1")&& !opcion.equals("a")&&!opcion.equals("2")&& !opcion.equals("b")
            &&!opcion.equals("3")&& !opcion.equals("c")){
        System.out.println("Opcion no valida");
    }
    while(!opcion.equals("s")&& !opcion.equals("S") && !opcion.equals(OPCION_SALIR)){
        System.out.println("Elige una opcion:\n\n1) o a)Rectangulo\n2) o b)Circunferencia\n"
                         + "3) o c)Triangulo\n\n 0) Salir>> ");

    opcion= keyboard.nextLine();
    switch(opcion){
        case OPCION_1,"a":
            
            System.out.println("LARGO >>");
            largo = keyboard.nextDouble();
            keyboard.nextLine();
            System.out.println("Ancho >>");
            ancho = keyboard.nextDouble();
            keyboard.nextLine();
        
            System.out.println("Perimetro = " + (largo * 2 + ancho * 2) 
                                + "Area= "+ (largo * ancho));
            break;
        case OPCION_2,"b":
            System.out.println("Radio =  ");
            radio = keyboard.nextDouble();
            keyboard.nextLine();
        
            System.out.println("Perimetro = "+ (radio * PI) + " Area = " 
                            + (PI* Math.pow(radio, 2)) );
            break;
            
        case OPCION_3,"c":
            System.out.println("Lado: ");
            lado = keyboard.nextDouble();
            keyboard.nextLine();
            System.out.println("Base: ");
            base = keyboard.nextDouble();
            keyboard.nextLine();
            System.out.println("Altura: ");
            altura = keyboard.nextDouble();
            keyboard.nextLine();
        
        
            System.out.println("Perimetro =" + (lado*2 + base) + "Area =" 
                            + (base*altura)/2);
            break;
            case OPCION_SALIR,"s","S":
            
            break;
        default:
            System.out.println("Error opcion no valida");
    
    }
    
            
    }
        
    }
}
